snapcraft/tests/test_sources.py

```diff
@@ -68,6 +101,102 @@ def test_pull_tarball_must_download_to_sourcedir(self, mock_prov):
         with open(os.path.join(dest_dir, tar_file_name), 'r') as tar_file:
             self.assertEqual('Test fake compressed file', tar_file.read())
 
+    def test_checksum(self):
+        tar = tarfile.open("checksum.tar", "w")
+        tar.close()
+
+        # md5
+        source_checksum = '1276481102f218c981e0324180bafd9f'
+        sources.verify_checksum(
+            source_checksum, "checksum.tar")
```

Comment:
Pushing a fix to a bunch of lines that can be fixed by this, thanks for pointing this out.